src/bootstrap/dist.rs

```diff
@@ -284,6 +284,50 @@ pub fn std(build: &Build, compiler: &Compiler, target: &str) {
     t!(fs::remove_dir_all(&image));
 }
 
+/// Creates a tarball of save-analysis metadata, if available.
+pub fn analysis(build: &Build, compiler: &Compiler, target: &str) {
+    println!("Dist analysis");
+
+    if build.config.channel != "nightly" {
+        println!("Skipping dist-analysis - not on nightly channel");
+        return;
+    }
+    if compiler.stage != 2 {
+        return
+    }
+
+    let name = format!("rust-analysis-{}", package_vers(build));
+    let image = tmpdir(build).join(format!("{}-{}-image", name, target));
+
+    let src = build.stage_out(compiler, Mode::Libstd).join(target).join("release").join("deps");
+
+    let image_src = src.join("save-analysis");
+    let dst = image.join("lib/rustlib").join(target).join("save-analysis");
```

Comment:
I'm not crazy about the name "save-analysis" here. The verb "save" sounds a bit wrong as the name of the folder containing the analysis.

What is the directory structure under "save-analysis"? I would expect that under this directory there are a series of directories named after target triples, since the analysis is per-target. Based on that assumption I would prefer if the analysis were stored under `lib/rustlib/$target/analysis` since we already have the organizational structure to distinguish targets under the rustlib dir. I'd also slightly prefer to put the analysis files side-by-side with the libraries they contain analysis for, directly under `lib/rustlib/$target/lib`, but can't articulate why, so maybe it doesn't matter.